src/user.ts

```diff
@@ -29,14 +29,32 @@ import { SecureBuffer } from './secureBuffer';
 import { SfdxError } from './sfdxError';
 import { sfdc } from './util/sfdc';
 
-const PASSWORD_LENGTH = 13;
-const LOWER = 'abcdefghijklmnopqrstuvwxyz';
-const UPPER = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
-const NUMBERS = '1234567890';
-const SYMBOLS = ['!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '[', ']', '|', '-'];
-
 const rand = (len: Many<string>): number => Math.floor(Math.random() * len.length);
 
+interface Complexity {
+  [key: string]: boolean | undefined;
+  LOWER?: boolean;
+  UPPER?: boolean;
+  NUMBERS?: boolean;
+  SYMBOLS?: boolean;
+}
+
+const CHARACTERS: { [index: string]: string | string[] } = {
+  LOWER: 'abcdefghijklmnopqrstuvwxyz',
+  UPPER: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+  NUMBERS: '1234567890',
+  SYMBOLS: ['!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '[', ']', '|', '-'],
+};
+
+const PASSWORD_COMPLEXITY: { [index: number]: Complexity } = {
+  '0': { LOWER: true },
+  '1': { LOWER: true, NUMBERS: true },
+  '2': { LOWER: true, SYMBOLS: true },
+  '3': { LOWER: true, UPPER: true, NUMBERS: true },
+  '4': { LOWER: true, NUMBERS: true, SYMBOLS: true },
+  '5': { LOWER: true, UPPER: true, NUMBERS: true, SYMBOLS: true },
```

Comment:
is 4 or 5 the most complex? We need to standardize on the server options

Reply:
Hmm, Vamsi originally had 4 and 5 switched (at least that's what he showed me when I worked on the description) and I asked the same question!  I think he switched it.  But did I steer him wrong?

Reply:
I think it makes more logical sense to have higher number = more security, but I thought the server had 4 as the most secure option... we need to make a note of what we're doing, and what the server is expecting

Reply:
@vamsimundra , please go with what the devs are saying!  Like I said earlier, to me the order you originally showed me looked wrong, which is why I flagged it, but I should really stay out of it.  But please make sure that whatever you decide is accurately reflected in the --help output.  Thanks